golem-cli/src/diagnose.rs

```diff
@@ -0,0 +1,893 @@
+                short_name: "wasi-sdk",
+                description: "WebAssembly toolchain for C and C++",
+                // NOTE: Version is not detectable currently, from 24.0 it will be stored in a version file
+                version_requirement: ExactByNameVersion("WASI_SDK set"),
+                instructions: indoc! {"
+                    Install WASI SDK 23.0:
+                        https://github.com/WebAssembly/wasi-sdk
+
+                    Don't forget to export the WASI_SDK environment variable!
+                "},
+            },
+            Tool::WasmTools => ToolMetadata {
+                short_name: "wasm-tools",
+                description: "Tools for manipulation of WebAssembly modules",
+                version_requirement: ExactVersion("1.210.0"),
+                instructions: indoc! {"
+                    Install the following specific version of wasm-tools:
+                        cargo install --force --locked  wasm-tools@1.210.0
+                "},
+            },
+            Tool::WitBindgen => ToolMetadata {
+                short_name: "wit-bindgen",
+                description: "Guest language bindings generator for WIT",
+                version_requirement: ExactVersion("0.26.0"),
+                instructions: indoc! {"
+                    Install the following specific version of wit-bindgen:
+                        cargo install --force --locked wit-bindgen-cli@0.26.0
+                "},
+            },
+            Tool::Zig => ToolMetadata {
+                short_name: "zig",
+                description: "Zig language tooling",
+                version_requirement: MinimumVersion("0.13.0"),
+                instructions: indoc! {"
+                    Install latest version of Zig:
+                        https://ziglang.org/learn/getting-started/#installing-zig
+                "},
+            },
+        }
+    }
+
+    pub fn direct_dependencies(&self) -> Vec<Tool> {
+        match self {
+            Tool::Cargo => vec![Tool::Rustup],
+            Tool::CargoComponent => vec![Tool::Cargo],
+            Tool::ComponentizeJs => vec![Tool::Npm],
+            Tool::ComponentizePy => vec![Tool::Pip],
+            Tool::Go => vec![],
+            Tool::GolemSdkGo => vec![Tool::Go],
+            Tool::GolemSdkRust => vec![Tool::Cargo],
+            Tool::GolemSdkTypeScript => vec![Tool::Npm],
+            Tool::Jco => vec![Tool::Npm],
+            Tool::Node => vec![],
+            Tool::Npm => vec![Tool::Node],
+            Tool::Pip => vec![Tool::Python],
+            Tool::Python => vec![],
+            Tool::Rustc => vec![Tool::Rustup],
+            Tool::Rustup => vec![],
+            Tool::RustTargetWasm32Wasi => vec![Tool::Rustc],
+            Tool::TinyGo => vec![Tool::Go],
+            Tool::WasiSdk => vec![],
+            Tool::WasmTools => vec![Tool::Cargo],
+            Tool::WitBindgen => vec![Tool::Cargo],
+            Tool::Zig => vec![],
+        }
```

Comment:
Very minor: I would have personally called it dependency-graph (because i see it as full dependency graph represented using adjacency list matrix strategy). 



It implies this adjacency-list is kept mainly to help out with looping through all dependencies in the graph and collect all of the dependencies. It would have been easier to understand if we kept the real graph structure as it is

Reply:
Ok, I see why you did this  (and named it that way) later down the line

Reply:
it is a simple dep list